iOS/MSMapsModules/MSMapsModules/MSMapGeoJsonParser/MSMapGeoJsonParser.m

```diff
@@ -0,0 +1,159 @@
+  layer = [[MSMapElementLayer alloc] init];
+  NSData *jsonData = [geojson dataUsingEncoding:NSUTF8StringEncoding];
+  NSError *jsonError;
+  NSDictionary *jsonObject =
+      [NSJSONSerialization JSONObjectWithData:jsonData
+                                      options:kNilOptions
+                                        error:&jsonError];
+
+  if (jsonObject == nil) {
+    *error = [NSError
+        errorWithDomain:@"com.microsoft.modules.MSMapsModules.JSONError"
+                   code:-200
+               userInfo:@{NSLocalizedDescriptionKey : @"Malformed JSON."}];
+    return nil;
+  }
+
+  NSString *type =
+      [NSString stringWithFormat:@"%@", [jsonObject objectForKey:@"type"]];
+  if ([type isEqualToString:@"FeatureCollection"]) {
+    // TODO: parseFeatureCollection
+  } else {
+    if ([type isEqualToString:@"Feature"]) {
+      // TODO: feature
+    }
+    [self switchToType:jsonObject error:error];
+  }
+
+  return layer;
+}
+
+- (void)switchToType:(NSDictionary *)object error:(NSError **)error {
+  NSString *type =
+      [NSString stringWithFormat:@"%@", [object objectForKey:@"type"]];
+  if ([type isEqualToString:@"Polygon"]) {
+    // TODO: parsePolygon
+  } else if ([type isEqualToString:@"Point"]) {
+    NSArray *coordinates = [self getCoordinates:object error:error];
+    if (coordinates != nil) {
+      [self parsePoint:coordinates error:error];
+    } else {
+      layer = nil;
+    }
+  }
+  // TODO: rest of geometry types
+}
+
+- (NSArray *)getCoordinates:(NSDictionary *)object error:(NSError **)error {
+  NSArray *coordinates = [object objectForKey:@"coordinates"];
+  if (coordinates == nil) {
+    *error = [NSError
+        errorWithDomain:@"com.microsoft.modules.MSMapsModules.JSONError"
+                   code:-200
+               userInfo:@{
+                 NSLocalizedDescriptionKey : @"Error getting coordinates array."
+               }];
+    return nil;
+  } else {
+    return coordinates;
+  }
+}
+
+- (void)parsePoint:(NSArray *)coordinates error:(NSError **)error {
+  MSGeoposition *position = [self parseGeoposition:coordinates error:error];
+
+  if (position == nil) {
+    layer = nil;
+    return;
+  }
+
+  MSGeopoint *point;
+  if (coordinates.count > 2) {
+    point = [[MSGeopoint alloc]
+               initWithPosition:position
+        altitudeReferenceSystem:MSMapAltitudeReferenceSystemEllipsoid];
+  } else {
+    point = [[MSGeopoint alloc]
+               initWithPosition:position
+        altitudeReferenceSystem:MSMapAltitudeReferenceSystemSurface];
+  }
+  MSMapIcon *icon = [[MSMapIcon alloc] init];
+  icon.location = point;
+  [layer.elements addMapElement:icon];
+}
+
+- (MSGeoposition *)parseGeoposition:(NSArray *)coordinates
+                              error:(NSError **)error {
+  if (coordinates.count >= 2) {
+    NSNumber *longitudeObj = coordinates[0];
+    double longitude = [longitudeObj doubleValue];
+    if (longitude < -180 || longitude > 180) {
+      // TODO: throw exception
```

Comment:
Populating error is probably a better approach. Please finish implementing this prior to checking in.